public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx

```diff
@@ -124,12 +126,14 @@ export const SharePublicDashboard = (props: Props) => {
             </div>
             <hr />
             <Configuration
+              annotationsEnabled={annotationsEnabled}
               dashboard={props.dashboard}
               disabled={!hasWritePermissions || isLoading || isFetchingError}
               isPubDashEnabled={enabledSwitch.isEnabled}
               onToggleEnabled={() =>
                 setEnabledSwitch((prevState) => ({ isEnabled: !prevState.isEnabled, wasTouched: true }))
               }
+              onAnnotationsToggled={() => setAnnotationsEnabled(!annotationsEnabled)}
```

Comment:
```suggestion
              onAnnotationsToggled={() => setAnnotationsEnabled(prevState => !prevState)}
```
because of the fact react state updates are async and queued (so not instantly executed) when you're switching the state based on a previous state, you shouldn't update it in this way, because there could be an scenario where you change it really quickly and the state was not updated yet, so you could have a bad behavior

Reply:
Interesting. How often does this actually happen? In all the time I've used React, I've never ran into that issue. Wouldn't the docs for `UseState` mention this if it was a problem?

Reply:
Here you have something
https://stackoverflow.com/questions/54807454/what-is-prevstate-in-reactjs
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous
https://stackoverflow.com/questions/40359800/how-to-toggle-boolean-state-of-a-react-component

Reply:
Neat, well I learned something new! Thanks for sharing. Will update 👍